docs/typescript/reference/dbos-class.md

```diff
@@ -0,0 +1,957 @@
+
+## Application Lifecycle
+For DBOS applications using entrypoint classes specified in [`dbos-config.yaml`](../reference/configuration), a "serverless" application lifecycle is handled automatically.  However, it is possible to build a custom application lifecycle.
+
+### Setting The Application Configuration
+If the DBOS configuration should not be loaded from `dbos-config.yaml`, it may be provided programatically with `DBOS.setConfig`:
+```typescript
+DBOS.setConfig(config: DBOSConfig, runtimeConfig?: DBOSRuntimeConfig)
+```
+
+If `runtimeConfig` is provided, the runtime will be started when the app is launched with `DBOS.launch`, including HTTP handling for decorated methods.  If `runtimeConfig` is not provided, this will not occur.
+
+### Launching and Shutting Down
+The DBOS app can be started with `launch` and stopped with `shutdown`.
+```typescript
+DBOS.launch(httpApps?: DBOSHttpApps)
+DBOS.shutdown()
+```
+
+* `launch`: This starts the DBOS app, or registers it to be started in conjunction with a provided HTTP server.  Launching will initialize all DBOS components, run any application initializers, start workflow recovery, start event processing, and start HTTP services (if so configured).
+* `shutdown`: This stops the DBOS app.  First, handling of new events and requests is disabled, and then the workflow executor and database connections are destroyed.
+
+### Recovering Workflows
+DBOS generally recovers workflows automatically.  However, this process can be initiated externally.
+
+```typescript
+DBOS.recoverPendingWorkflows(executorIDs: string[] = ["local"]): Promise<WorkflowHandle<unknown>[]>
+DBOS.executeWorkflowById(workflowId: string, startNewWorkflow: boolean = false): Promise<WorkflowHandle<unknown>>
+```
+
+* `DBOS.recoverPendingWorkflows`: Retrieves a list of workflows to recover, and starts them, returning their handles
+* `DBOS.executeWorkflowById`: Starts executing a workflow given its `workflowId`.  If `startNewWorkflow` is `true`, the workflow is cloned and assigned a new unique workflow ID.
+
+## Extensibility
+DBOS Transact allows libraries, such as those designed to receive external events, to initiate DBOS functions.
+
+### Decorators
+Such libraries may choose to implement method decorators.  If the method decorator is registering a function with DBOS Transact, it should call `DBOS.registerAndWrapContextFreeFunction` for functions that do not accept a context as the first parameter.
+
+```typescript
+DBOS.registerAndWrapContextFreeFunction<This, Args extends unknown[], Return>(
+    target: object,
+    propertyKey: string,
+    descriptor: TypedPropertyDescriptor<(this: This, ...args: Args) => Promise<Return>>,
+)
+```
+
+`DBOS.registerAndWrapContextFreeFunction` registers the function named `propertyKey` on the `target` class, creates a wrapper function, and updates the `descriptor` to use the wrapper function.
```

Comment:
The user-facing name could be shorter, like `DBOS.createDecorator` or so...

Reply:
See above - I think when we try this we will change it a bit.  `createDecorator` is not right, though.

Reply:
Cut this section for now.